From d744317e90dfc012fec61b3b000b66026c5c8c81 Mon Sep 17 00:00:00 2001
From: Andreas Gampe <agampe@google.com>
Date: Mon, 22 Jan 2018 21:24:48 -0800
Subject: [PATCH 07/68] Gallery2: Fix NaN comparisons

Use Float.isNan for correct evaluation. Found by errorprone.

Test: m
Change-Id: I3d47a2a64059d9c89528c82a1ca766248db4bdf9
---
 .../filtershow/filters/FilterVignetteRepresentation.java    | 2 +-
 .../gallery3d/filtershow/imageshow/EclipseControl.java      | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/com/android/gallery3d/filtershow/filters/FilterVignetteRepresentation.java b/src/com/android/gallery3d/filtershow/filters/FilterVignetteRepresentation.java
index cff923132..e90e12fb4 100644
--- a/src/com/android/gallery3d/filtershow/filters/FilterVignetteRepresentation.java
+++ b/src/com/android/gallery3d/filtershow/filters/FilterVignetteRepresentation.java
@@ -151,7 +151,7 @@ public class FilterVignetteRepresentation extends FilterRepresentation implement
     }
 
     public boolean isCenterSet() {
-        return mCenterX != Float.NaN;
+        return !Float.isNaN(mCenterX);
     }
 
     @Override
diff --git a/src/com/android/gallery3d/filtershow/imageshow/EclipseControl.java b/src/com/android/gallery3d/filtershow/imageshow/EclipseControl.java
index 3e6d0e71f..b67da4633 100644
--- a/src/com/android/gallery3d/filtershow/imageshow/EclipseControl.java
+++ b/src/com/android/gallery3d/filtershow/imageshow/EclipseControl.java
@@ -189,7 +189,7 @@ public class EclipseControl {
     }
 
     public void paintGrayPoint(Canvas canvas, float x, float y) {
-        if (x == Float.NaN) {
+        if (Float.isNaN(x)) {
             return;
         }
 
@@ -206,7 +206,7 @@ public class EclipseControl {
     }
 
     public void paintPoint(Canvas canvas, float x, float y) {
-        if (x == Float.NaN) {
+        if (Float.isNaN(x)) {
             return;
         }
 
@@ -223,7 +223,7 @@ public class EclipseControl {
     }
 
     void paintRadius(Canvas canvas, float cx, float cy, float rx, float ry) {
-        if (cx == Float.NaN) {
+        if (Float.isNaN(cx)) {
             return;
         }
         int mSliderColor = 0xFF33B5E5;
-- 
2.17.1

